examples/react-spa/src/sdk.ts

```diff
@@ -2,15 +2,142 @@
 // SPDX-License-Identifier: Apache-2.0
 
 import { Configuration, FrontendApi } from "@ory/client"
+import { AxiosError } from "axios"
+import React, { useCallback } from "react"
+import { useNavigate } from "react-router-dom"
 
-export default new FrontendApi(
+export const sdk = new FrontendApi(
   new Configuration({
     //https://vitejs.dev/guide/env-and-mode.html#env-files
-    basePath: import.meta.env.VITE_ORY_SDK_URL,
+    basePath: "https://kratos.climbeyond.app",
     // we always want to include the cookies in each request
     // cookies are used for sessions and CSRF protection
     baseOptions: {
       withCredentials: true,
     },
   }),
 )
+
+/**
+ * @param getFlow - Should be function to load a flow make it visible (Login.getFlow)
+ * @param setFlow - Update flow data to view (Login.setFlow)
+ * @param defaultNav - Default navigate target for errors
+ * @param fatalToDash - When true and error can not be handled, then redirect to dashboard, else rethrow error
+ */
+export const sdkError = (
+  getFlow: ((flowId: string) => Promise<void | AxiosError>) | undefined,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  setFlow: React.Dispatch<React.SetStateAction<any>> | undefined,
+  defaultNav: string | undefined,
+  fatalToDash = false,
+) => {
+  const navigate = useNavigate()
+
+  return useCallback(
+    (error: AxiosError): Promise<AxiosError | void> => {
+      const responseData = error.response?.data || {}
+
+      switch (error.response?.status) {
+        case 400: {
+          // the request could contain invalid parameters which would set error messages in the flow
+          if (setFlow !== undefined) {
+            console.warn("sdkError 400: update flow data")
+            setFlow(responseData)
+            return Promise.resolve()
+          }
+          break
+        }
+        case 401: {
+          console.warn("sdkError 401: Navigate to /login")
+          navigate("/login", { replace: true })
+          return Promise.resolve()
+        }
+        case 403: {
+          // the user might have a session, but would require 2FA (Two-Factor Authentication)
+          if (responseData.error?.id === "session_aal2_required") {
+            navigate("/login?aal2=true", { replace: true })
+            return Promise.resolve()
+          }
+
+          if (
+            responseData.error?.id === "session_refresh_required" &&
+            responseData.redirect_browser_to
+          ) {
+            console.warn("sdkError 403: Redirect browser to")
+            window.location = responseData.redirect_browser_to
+            return Promise.resolve()
+          }
+          break
+        }
+        case 404: {
+          if (defaultNav !== undefined) {
+            console.warn("sdkError 404: Navigate to", defaultNav)
+            navigate(defaultNav, { replace: true })
```

Comment:
With a 404 it means the route isn't known on the SDK or the SDK was misconfigured. You could redirect the user to the error page here with your own error now stating that. I think it's a good solution for first time users of Ory to have a self-diagnosing example.

Reply:
Must have been able to break some flow links to browser history because I got 404 messages from there too and that redirect is broken so the error page is definitely a better way to handle it.